lib/improver/weights.py

```diff
@@ -233,29 +377,72 @@ def __init__(self, cval=0.85):
         """
         self.cval = cval
 
-    def process(self, cube, coord):
+    def nonlinear_weights(self, num_of_weights):
+        """Create nonlinear weights.
+
+            Args:
+                num_of_weights : Positive Integer
+                                 Number of weights to create.
+
+            Returns:
+                weights : array of weights, sum of all weights = 1.0
+
+            Raises:
+                ValueError: an inappropriate value of cval is input.
+        """
+        if self.cval <= 0.0 or self.cval > 1.0:
+            msg = ('cval must be greater than 0.0 and less '
+                   'than or equal to 1.0 '
+                   'cval = {0:s}'.format(str(self.cval)))
+            raise ValueError(msg)
+
+        weights_list = []
+        for tval_minus1 in range(0, num_of_weights):
+            weights_list.append(self.cval**(tval_minus1))
+
+        weights = WeightsUtilities.normalise_weights(np.array(weights_list))
+
+        return weights
+
+    def process(self, cube, coord_name, coord_vals=None, coord_unit='no_unit',
+                weights_distrib_method='evenly'):
         """Calculated weights for a given cube and coord.
 
             Args:
                 cube : iris.cube.Cube
                        Cube to blend across the coord.
-                coord : string
-                        The name of a coordinate dimension in the cube.
+                coord_name : string
+                       Name of coordinate in the cube to be blended.
+                coord_vals : string
+                       String list of values which are expected on the
+                       coordinate to be blended over.
+                coord_unit : cf_units.Unit
+                       The unit in which the coord_exp_vals have been passed
+                       in.
+                weights_distrib_method : string
+                        The method to use when redistributing weights in cases
+                        where there are some forecasts missing. Options:
+                        "evenly", "proportional".
```

Comment:
This last description is indented one more space than those above. :space_invader: :space_invader: